### PR TITLE
Reformulate second order pattern detection

### LINKED ISSUE #35: Nonlinear pattern detection
- Replace the verb "involve"

### PR DESCRIPTION
**Pull request by adrhill**
Closes #35.

### CODE REVIEW

File: _posts/2025-04-28-sparse-autodiff.md [hunk 1]
```diff
@@ -699,13 +697,16 @@ The complexity of a single HVP scales roughly with the complexity of the functio
...
-For instance, if $f(\mathbf{x})$ involves a term of the form $x_1 + x_2$, it will not affect the Hessian. 
-On the other hand, a term $x_1 x_2$ will yield two equal non-zero coefficients, one at position $(1, 2)$ and one at position $(2, 1)$.
-Thus, the abstract interpretation system used for detection needs a finer classification of operators, to distinguish between locally linear ones (sum, max) and locally nonlinear ones (product, exp).
+This is because, in addition to the usual linear dependencies, we now have to account for **nonlinear interactions** in the compute graph.
+
+For instance, if $f(\mathbf{x})$ involves a term of the form $x_1 + x_2$, it will not directly affect the Hessian. 
+On the other hand, multiplying this term with $x_3$ to obtain an output $f(\mathbf{x}) = (x_1 + x_2)\,x_3$ 
+will yield non-zero coefficients at position $(1, 3)$, $(3, 1)$, $(2, 3)$ and $(3, 2)$.
+Thus, the abstract interpretation system used for detection needs a finer classification of operators, 
+distinguishing between locally constant, locally linear, and locally nonlinear behavior in each directional derivative.
```
Comment (gdalle) [inline _posts/2025-04-28-sparse-autodiff.md:709]:
```suggestion
distinguishing between locally constant, locally linear, and locally nonlinear behavior. 
```

Comment (adrhill) [reply _posts/2025-04-28-sparse-autodiff.md:709]: I want to hint at the fact that 3 classes are not enough. There are 13 types of different binary operators:
https://github.com/adrhill/SparseConnectivityTracer.jl/blob/e6d357b5e0b23d72ab3430e359831eafeb005d73/src/operators.jl#L389-L410

Comment (gdalle) [inline _posts/2025-04-28-sparse-autodiff.md:709]:
```suggestion
distinguishing between locally constant, locally linear, and locally nonlinear behavior.
To further complicate things, an operator may be linear with respect to an argument and nonlinear with respect to another (like $(x, y) \mapsto x / y$). 
```

Comment (gdalle) [reply _posts/2025-04-28-sparse-autodiff.md:709]: I thought you meant "each directional derivative" as "each HVP". I suggested another sentence to clarify this

Comment (gdalle) [reply _posts/2025-04-28-sparse-autodiff.md]: Dude just pick something and merge ^^

**Pull request merged**